### ISSUE TITLE
Basic Help

### ISSUE THREAD

**Issue by SeanFairhurst** [suytt/esri.symbol.MultiLineTextSymbol#1]
I'm having trouble getting your examples to work.  I see the text in the middle of the map and the \n does not show but it doesn't produce a new line either.  Can you help?

**Comment by suytt** [comment]
Can you please specify which browser and version you are using ?

**Comment by SeanFairhurst** [comment]
I tried IE9 and chrome   Thanks for your time on this, I tried to point it
v3.7 of ESRI's javascript api as well.

Sean Fairhurst

Chief Technology Officer

[image: cai-tech.com] http://www.cai-tech.com

Toll-free 800.322.4540 x12

cai-tech.com http://www.cai-tech.com

  http://www.linkedin.com/company/cai-technologies?trk=company_logo

On Tue, May 20, 2014 at 1:59 AM, Simon Uyttendaele <notifications@github.com

**Comment by suytt** [comment]
I've set a default line height which was causing overlapping lines in chrome.
However, in IE9, IE11, FF and chrome, it works for me.
Can you eventually send a screenshot ?

**Comment by pshahid** [comment]
Chaussette thanks for providing this patch! I'm having trouble getting multi-line text to show as well, same problem as described by Sean. I'm using Esri JS API v3.9, and have confirmed this on IE11, Chrome (v 35.0.1916.114) and FF (v 29.0.1). 

What I have noticed by inspecting the HTML elements that get put on the map is that they appear correctly displayed in the raw HTML but the way they're rendered by browsers is just 1 line. Here's an imgur album of an example of what I'm talking about: http://imgur.com/a/ru9nT.

I have confirmed that I'm using the latest version of MultiLineTextSymbol.js. 

EDIT: Disregard the multiple returns there, I split the text string on spaces and replace them with new line characters like this: " \n "

**Comment by suytt** [comment]
Thanks for the feedback, I'll give v3.9 a try.
The dojo version bundled with the api may differ and this fix may thus need an update.

Stay tuned for the updates.

**Comment by suytt** [comment]
@pshahid : I am not able to reproduce the issue with a simple example and v3.9. Thus I believe it may come from timing issues or conflicts with other parts of the code.
Is it possible for you to send me (part of) the code and/or the url to the page so that I can try to figure out what is happning ?

**Comment by pshahid** [comment]
@Chaussette I appreciate your quick response!  Unfortunately, I can't share the code because it's proprietary and I'm not allowed to divulge pretty much anything, but I'm working on a snippet that I can send you. 

In the mean-time, I noticed that, at least with our map, the text symbol would not show up unless you gave a spatial reference to the point (if you added it through the map.graphics.add way).

The way we're doing this for production, though, is a custom-made layer that is built on top of Esri's GraphicsLayer. Regardless of which way the symbol is added it doesn't break into newlines as it should though, that's for sure.

**Comment by pshahid** [comment]
Alright. I am really sorry for wasting your time.

I fixed it and here's what was happening with: there was an angular bracket missing on the script tag for the include. This manifested without any errors in the console and of course the HTML renderers just absorbed the problem and moved on. The text symbol was being created with the Esri TextSymbol instead of this library. @SeanFairhurst this may be happening with you as well, or some derivative error thereof. 

TL;DR: I am dumb

**Comment by suytt** [comment]
Well, that happens to all of us :)
Good to know that it finally works for you then.

**Comment by saclarky** [comment]
This patch is such a help, thanks so much for your time on it. I've noticed when I upgrade my apps to 3.13 it is no longer working. Any thoughts? The code is currently a little over my head.

**Comment by suytt** [comment]
Hello @saclarky 
I've just tester using v3.13 and it works like a charm for me.
Do you have a sample page I could look at to try to help you ? 
What browser + version are you using ?

**Comment by saclarky** [comment]
I unfortunately can't share my app but I was able to replicate it using this ESRI sample: http://developers.arcgis.com/javascript/sandbox/sandbox.html?sample=layers_label

I added the patch from: https://rawgit.com/Chaussette/esri.symbol.MultiLineTextSymbol/master/esri.symbol.MultiLineTextSymbol.js

I changed Line 68 to: labels.addFeatureLayer(states, statesLabelRenderer, "test" + "\n" + "{" + labelField + "}");

I am in Chrome 41

**Comment by suytt** [comment]
Ok, I see, 
The problem is that esri created a new type of layer: LabelLayer which seems to strip the newline character somewhere in the process.
I'll have a loot at it

**Comment by suytt** [comment]
@saclarky : voila, the sandbox example works like a charm. If you encounter other problems, please let me know.

**Comment by saclarky** [comment]
Looks good to me, thanks a million @suytt
